### PR TITLE
chore: Add features from vector-aggregator chart

### LINKED ISSUE #35: Ensure feature parity between vector and vector-aggregator
Ensure adequate test coverage with `**/ci/*-values.yaml` files

From #24

### PR DESCRIPTION
**Pull request by spencergilbert**
- feat(vector): Add dnsConfig and dnsPolicy support
- feat(vector): Add support for an existing ConfigMap
- feat(vector): Add support for termination period and management policy
- feat(vector): Add PodMonitor and update default prometheus port name
- feat(vector): Support existing ConfigMap for HAProxy

Closes #35

I still need to determine what options we should exercise in CI for the Aggregator.

### CODE REVIEW

**Comment by spencergilbert** [comment]
I _think_ I'll update the `ci` files to be `<role>-<"purpose">-values.yaml` like `aggregator-customConfig-values.yaml`

File: charts/vector/templates/haproxy/configmap.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.haproxy.enabled }}
+{{- if and .Values.haproxy.enabled (not .Values.existingConfigMap) }}
```
Comment (StephenWakely) [inline charts/vector/templates/haproxy/configmap.yaml:1]: Does `not` on an empty string return `true`?

Comment (spencergilbert) [reply charts/vector/templates/haproxy/configmap.yaml:1]: 👍  empty is ~= false when evaluating templates

File: charts/vector/templates/_pod.tpl [hunk 2]
```diff
@@ -73,11 +80,11 @@ containers:
       - name: vector
         containerPort: 6000
         protocol: TCP
-      - name: prometheus
+      - name: prom-exporter
```
Comment (StephenWakely) [inline charts/vector/templates/_pod.tpl:83]: Here `prometheus` is being renamed to `prom-exporter`, but below `prometheus-exporter` is being renamed to `prom-exporter`. Is it ok that things that had different names now have the same name?

Comment (spencergilbert) [reply charts/vector/templates/_pod.tpl:83]: So I found out that `containerPort` names can't be more than 15(16?) characters, which broke when autogenerating them based off `customConfig` - I wanted to consistently name things across `Pods` and `Services` (as well as the provided defaults and what the generated output is.

tl;dr I wanted to ensure that if you passed what I've provided as default into the `customConfig` option you'd get the same output as just using the defaults.

**Review by StephenWakely — APPROVED** [review]
Looks good to me!

File: charts/vector/templates/NOTES.txt [hunk 3]
```diff
@@ -12,7 +12,7 @@ Vector is starting in your cluster. After a few minutes, you can use Vector's AP
   {{- else if .Values.customConfig.api.enabled }}
 Vector is starting in your cluster. After a few minutes, you can use Vector's API to view internal metrics by running:
 
-	kubectl -n {{ .Release.Namespace }} exec -it statefulset/{{ include "vector.fullname" . }} -- vector top --url {{ .Values.customConfig.api.address | default "http://127.0.0.1:8686/graphql" }}
+	kubectl -n {{ .Release.Namespace }} exec -it statefulset/{{ include "vector.fullname" . }} -- vector top --url {{ printf "http://%s/graphql" .Values.customConfig.api.address | default "http://127.0.0.1:8686/graphql" }}
```
Comment (spencergilbert) [inline charts/vector/templates/NOTES.txt:15]: This command would fail if the `api.address` was replaced in `customConfig` as we didn't pass the full url and path

**Review by jszwedko — APPROVED** [review]
Looks good! Thanks for keeping the commits specific.
